CMakeLists.txt

```diff
@@ -163,6 +163,13 @@ ADD_CUSTOM_TARGET(setup_tests
   WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
   COMMENT "Enabling all tests ...")
 
+# disable the ability to run all tests in an in-source build:
+IF (("${CMAKE_SOURCE_DIR}" STREQUAL "${CMAKE_BINARY_DIR}") AND ASPECT_RUN_ALL_TESTS)
+  MESSAGE(WARNING "\nEnabling all tests is not supported in in-source builds. Please create a separate build directory!\n")
```

Comment:
Just make it FATAL and not worry about disabling the variable in the next line.

Reply:
Where is `ASPECT_RUN_ALL_TESTS` set, by the way? When you do `make setup_tests`, or also otherwise?

Reply:
It s a cmake flag you can set (``cmake -D ASPECT_RUN_ALL_TESTS=ON``) or it is enabled by running ``make setup_tests``. I am not sure failing here would be useful as this will make ``make setup_tests`` fail, but I can change it if you prefer.

Reply:
Yes, please. Silently doing something other than the person intended is not nice :-)